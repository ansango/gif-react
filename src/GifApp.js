import { useState } from "react";
import AddCategory from "./components/AddCategory";
import { Container, ContainerTitle } from "./components/Container";
import GifGrid from "./components/GifGrid";
import GifTitle from "./components/GigTitle";

const GifApp = ({ defaultCategories = []}) => {
  const [categories, setCategories] = useState(defaultCategories);

  return (
    <Container>
      <ContainerTitle>
        <GifTitle title={"GIPHY React App"} />
      </ContainerTitle>
      <AddCategory setCategories={setCategories} />
      <hr />
      {categories.map((category) => (
        <GifGrid key={category} category={category} />
      ))}
      {categories.length === 0 && (
        <ContainerTitle>
          <h2 className="font-bold text-2xl py-5">
            Enter a value to start a new search
          </h2>
        </ContainerTitle>
      )}
    </Container>
  );
};

export default GifApp;
