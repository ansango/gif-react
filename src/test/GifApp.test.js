import { shallow } from "enzyme";
import GifApp from "../GifApp";
describe("Test <GifApp/>", () => {
  test("should render", () => {
    const wrapper = shallow(<GifApp />);
    expect(wrapper).toMatchSnapshot();
  });

  test("should render a categories list", () => {
    const categories = ["One Punch", "Dragon Ball"];
    const wrapper = shallow(<GifApp defaultCategories={categories} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("GifGrid").length).toBe(categories.length);
  });
});
