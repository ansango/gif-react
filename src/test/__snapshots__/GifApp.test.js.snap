// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <GifApp/> should render 1`] = `
<Container>
  <ContainerTitle>
    <GifTitle
      title="GIPHY React App"
    />
  </ContainerTitle>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ContainerTitle>
    <h2
      className="font-bold text-2xl py-5"
    >
      Enter a value to start a new search
    </h2>
  </ContainerTitle>
</Container>
`;

exports[`Test <GifApp/> should render a categories list 1`] = `
<Container>
  <ContainerTitle>
    <GifTitle
      title="GIPHY React App"
    />
  </ContainerTitle>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <GifGrid
    category="One Punch"
    key="One Punch"
  />
  <GifGrid
    category="Dragon Ball"
    key="Dragon Ball"
  />
</Container>
`;
